src/sentry/scripts/similarity/index.lua

```diff
@@ -789,161 +708,96 @@ local commands = {
-        function (configuration, arguments)
-            local entries = build_variadic_argument_parser({
-                {'index', identity},
-                {'key', identity},
-                {'data', cmsgpack.unpack},
-            })(arguments)
-
-            for _, entry in ipairs(entries) do
-                for band, data in ipairs(entry.data) do
-                    for _, item in ipairs(data) do
-                        local time, buckets = item[1], item[2]
-                        local expiration_time = get_index_expiration_time(
-                            configuration.interval,
-                            configuration.retention,
-                            time
-                        )
-                        local destination_bucket_frequency_key = get_bucket_frequency_key(
-                            configuration,
-                            entry.index,
-                            time,
-                            band,
-                            entry.key
-                        )
-
-                        for bucket, count in pairs(buckets) do
-                            local bucket_membership_key = get_bucket_membership_key(
-                                configuration,
-                                entry.index,
-                                time,
-                                band,
-                                bucket
-                            )
-                            redis.call('SADD', bucket_membership_key, entry.key)
-                            redis.call('EXPIREAT', bucket_membership_key, expiration_time)
-
-                            redis.call(
-                                'HINCRBY',
-                                destination_bucket_frequency_key,
-                                bucket,
-                                count
-                            )
-                        end
-
-                        -- The destination bucket frequency key may have not
-                        -- existed previously, so we need to make sure we set
-                        -- the expiration on it in case it is new. (We only
-                        -- have to do this if there we changed any bucket counts.)
-                        if next(buckets) ~= nil then
-                            redis.call(
-                                'EXPIREAT',
-                                destination_bucket_frequency_key,
-                                expiration_time
-                            )
-                        end
+        local cursor, entries = variadic_argument_parser(
+            object_argument_parser({
+                {'index', argument_parser(validate_value)},
+                {'key', argument_parser(validate_value)},
+                {'frequencies', argument_parser(cmsgpack.unpack)}
+            })
+        )(cursor, arguments)
+
+        return table.imap(
+            entries,
+            function (source)
+                -- TODO: Restore the TTL here?
```

Comment:
This will still have a TTL set but is probably further in the future than necessary since it's relative to the script execution time and not the original source data TTL